lib/Core/BuildEngine.cpp

```diff
@@ -109,9 +110,13 @@ class BuildEngineImpl : public BuildDBDelegate {
     /// Force the use of a prior value
     bool forcePriorValue = false;
   };
-  std::vector<TaskInputRequest> inputRequests;
+  std::deque<TaskInputRequest> inputRequests;
```

Comment:
Should we worry about this having a perf impact?

Reply:
Existing perf tests show +/- 3-5%, but those can go in either direction in repeated test runs.

Reply:
Cool, thanks for checking.